### DISCUSSION TITLE
is there a way to change the property names used for exception handlers?

### DISCUSSION THREAD

**Discussion by scott2b** [litestar-org/litestar#787] [Q&A] [answered]
I am wondering if something like this is possible: expose the property on a 400 response that is normally called `status_code` as e.g. `status` or `statusCode`.

For my own validators, I am experimenting with pydantic's `alias_generator`. I have a camel-case converter, and then place a Config on my models like so:

```
    class Config:
        alias_generator = to_camel
        allow_population_by_field_name = True
```

So far, this appears to do what I am expecting. It exposes the validation properties as camelCase, while allowing me to keep my pythonic snake_case properties internally. There are a few things I have to do to make this work:

1. Define the above mentioned config on all my input and output validators
2. In the controller, validate the response with the response validator, and then return the result of `.dict(by_alias=True)`
3. ~Define incoming parameters with the `Parameter` class so that I can specify the camel cased name of the property. Path string matchers in the route decorators then use the camel case version of the property name.~

So far this looks good but then I have lingering things like the status_code on 400 responses or other public exceptions. From what I can tell, these don't really go through validation and the given properties are just coded into the HTTPException. So I'm not sure if there is really a straightforward way to change this.

**Comment by scott2b** [comment]
Update. I guess point 3 above does not work like I thought. I suppose I would need to specify camel case path parameters. I don't see a way to specify a path parameter alias the way you can a query parameter. I was under the impression that the query property was overloaded for both, but I am seeing now that does not seem to be the case.

**Comment by Goldziher** [comment, marked as answer]
You can define a custom serializer on your response and modify the data there.
